lib/chef/knife/yaml_convert.rb

```diff
@@ -0,0 +1,87 @@
+#
+# Author:: Bryan McLellan <btm@loftninjas.org>
+# Copyright:: Copyright 2020, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "yaml"
+require_relative "../knife"
+class Chef::Knife::YamlConvert < Chef::Knife
+
+  banner "knife yaml convert YAML_FILENAME [RUBY_FILENAME]"
+
+  def run
+    if name_args.empty?
+      ui.fatal!("Please specify the file name of a YAML recipe to convert to Ruby")
+    elsif name_args.size >= 3
+      ui.fatal!("knife yaml convert YAML_FILENAME [RUBY_FILENAME]")
+    end
+
+    yaml_file = @name_args[0]
```

Comment:
It would be nice if we can check the existing of the provided file here.

```suggestion
    yaml_file = @name_args[0]
    ui.fatal!("File #{yaml_file} does not exist/unadle to read") unless ::File.exist?(yaml_file)
```